app/controllers/logs_controller.rb

```diff
@@ -1,7 +1,14 @@
 class LogsController < ApplicationController
   def index
-    @logs = UseCases::Administrator::GetAuthRequestsForUsername.new(
+    auth_requests = UseCases::Administrator::GetAuthRequestsForUsername.new(
       authentication_logs_gateway: Gateways::LoggingApi.new
     ).execute(username: params[:username])
+
+    filter_auth_requests_by_organisation(auth_requests)
+  end
+
+  def filter_auth_requests_by_organisation(auth_requests)
+    ips = current_organisation.ips.map(&:address)
+    @logs = auth_requests.select { |auth_request| ips.include?(auth_request["siteIP"]) }
```

Comment:
I don't like setting the instance variables in a method like this - both from a clarity and "rails convention" point-of-view.

I'd just rejig things slightly so the `index` method doesn't have any side-effects:

```ruby
def index
    auth_requests = UseCases::Administrator::GetAuthRequestsForUsername.new(
      authentication_logs_gateway: Gateways::LoggingApi.new
    ).execute(username: params[:username])

    @logs = filter_by_organisation(auth_requests)
  end

  def filter_by_organisation(auth_requests)
    ips = current_organisation.ips.map(&:address)
    auth_requests.select { |auth_request| ips.include?(auth_request["siteIP"]) }
  end
end
```
(didn't test this)

This whole controller is very small at the moment, so this is more of a future concern, but I'd fix it now.

Reply:
Yeah I tend to agree, if you are setting a crucial instance variable for the view eg `@logs` then they should be set in the index action.

But I think you'll fix this in the merge conflict anyway George.